examples/user_guide/Linking_Plots.ipynb

```diff
@@ -0,0 +1,220 @@
+    "\n",
+    "We can once again look at the handles on the plot corresponding to the ``HLine`` element:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot = renderer.get_plot(hv.HLine(0))\n",
+    "plot.handles.keys()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now want to change the ``glyph``, which defines the position of the ``HLine``, so we declare the ``target_model`` as ``'glyph'``. Having defined both the source and target model and the events we can finally start writing the JS callback that should be triggered. To declare it we simply define the ``source_code`` class attribute. To understand how to write this code we need to understand how the source and target models we have declared can be referenced from within the callback.\n",
+    "\n",
+    "The ``source_model`` will be made available by prefixing it with ``source_``, while the target model is made available with the prefix ``target_``. This means that the ``ColumnDataSource`` on the ``source`` can be referenced as ``source_source``, while the glyph on the target can be referenced as ``target_glyph``.\n",
+    "\n",
+    "Once we know how to reference the models we can access their properties to compute the mean y-value of the current selection on the source ``ColumnDataSource`` and set the ``target_glyph.position`` to that value:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from holoviews.plotting.bokeh.callbacks import LinkCallback\n",
+    "\n",
+    "class MeanHLineCallback(LinkCallback):\n",
+    "\n",
+    "    source_model = 'source'\n",
+    "    on_source_changes = ['selected']\n",
+    "    \n",
+    "    target_model = 'glyph'\n",
+    "    \n",
+    "    source_code = \"\"\"\n",
+    "        var inds = source_source.selected.indices;\n",
+    "        var d = source_source.data;\n",
+    "        var ym = 0\n",
+    "        if (inds.length == 0)\n",
+    "            return;\n",
+    "        for (var i = 0; i < inds.length; i++)\n",
+    "            ym += d['y'][inds[i]]\n",
+    "        ym /= inds.length\n",
+    "        target_glyph.location = ym\n",
+    "    \"\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally we need to register the ``MeanHLineCallback`` with the ``MeanHLineLink`` via the ``Link._callbacks`` dictionary:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Link._callbacks['bokeh'][MeanHLineLink] = MeanHLineCallback"
```

Comment:
Would be nice to have a better API for registering these things together than using ``_callbacks`` directly like this.

Reply:
True, Streams should probably mirror that API since it's identical.